### PR TITLE
Issue/create user 

### PR DESCRIPTION
**Pull request by rachel-labri-tipton**
# Pull Request

## Description

Adds a section to the OCF Dashboard that adds a user. 
There's a minimal check that an email is formatted correctly. 

#TODO create error message when user already exists

Screenshot 2023-10-12 at 11 09 01

Screenshot 2023-10-12 at 11 12 27

## How Has This Been Tested?

Wrote a test for the function `create user`
Ran the code locally

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/get_data.py [hunk 1]
```diff
@@ -271,3 +282,31 @@ def create_new_site(
...
+        .filter(SiteGroupSQL.site_group_name == site_group_name)
+        .first()
+    )
+
+    user = UserSQL(email=email, site_group_uuid=site_group.site_group_uuid)
+
+    session.add(user)
+    site_group.users.append(user)
+    session.commit()
+    print(user)
```
Comment (peterdudfield) [inline src/get_data.py:310]: remove print

File: src/sites_toolbox.py [hunk 2]
```diff
@@ -372,3 +403,41 @@ def sites_toolbox_page():
...
+                    st.markdown(f'<p style="color:#f07167;font-size:16px;">{"Please enter a valid email address."}</p>',
+                                unsafe_allow_html=True,)
+                else:
+                    site_group = get_site_group_by_name(
+                        session=session, site_group_name=site_group
+                    )
+                    user = create_user(
+                        session=session,
+                        email=email,
+                        site_group_name=site_group.site_group_name,
```
Comment (peterdudfield) [inline src/sites_toolbox.py:434]: Seems like you have the `site_group_name`, then you get the site_group, and then get the name from it. 

perhaps further up `site_group` should be renamed `site_group_name` to make it clear

why dont use just pass `site_group_name` into this, seems

File: tests/test_get_data.py [hunk 3]
```diff
@@ -95,3 +96,18 @@ def test_create_new_site_twice(db_session):
     assert site_1.ml_id == 1
     assert site_2.ml_id == 2
 
+
+def test_create_user(db_session):
+    "Test to create a new user."
+
+    site_group_1 = make_site_group(db_session=db_session)
```
Comment (peterdudfield) [inline tests/test_get_data.py:103]: nice test

**Comment by peterdudfield** [comment]
Nice work @rachel-labri-tipton , ive added a few comments which are small. 
And yea, would be good to do that error message as you have put as a TODO

File: src/sites_toolbox.py (hunk 2 above)
Comment (rachel-labri-tipton) [reply src/sites_toolbox.py:444]: i think there's something funny with an SQL Alchemy object being returned and throwing an error, but I should be able to clean up the code so it's clearer.

Comment (peterdudfield) [reply src/sites_toolbox.py:444]: ok, let me know if you need any help on that

File: src/sites_toolbox.py [hunk 4]
```diff
@@ -434,19 +434,16 @@ def sites_toolbox_page():
...
-                    site_group = get_site_group_by_name(
-                        session=session, site_group_name=site_group_name
-                    )
 
                     user = create_user(
                         session=session,
                         email=email,
-                        site_group_name=site_group.site_group_name,
+                        site_group_name=site_group_name,
                     )
```
Comment (rachel-labri-tipton) [inline src/sites_toolbox.py:443]: @peterdudfield, this commit should fix the confusion between `site_group` and `site_group_name`

Comment (peterdudfield) [reply src/sites_toolbox.py:443]: nice, and I can see you have removed a call to get site group

**Review by peterdudfield — APPROVED** [review]

**Pull request closed without merging**